### PR TITLE
Setup Repo for deployment

### PR DESCRIPTION
**Pull request by jhbertra**
- [x] Refactors nix to use `std`(along with README updates)
- [x] Updates project to use libraries published on CHaP
- [x] Adds `operable` and `oci-image` build targets for the frontend and the backend
- [x] Adds GitHub Actions workflows to publish images on pushes to `main` and version tags.
- [ ] TODO deploy new images to KubeVela (requires Flux CI, which is currently in testing).

### CODE REVIEW

File: .envrc [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+source "$(
+  nix eval \
+  --no-update-lock-file \
+  --no-write-lock-file \
+  --no-warn-dirty \
+  --accept-flake-config \
+  .#__std.direnv_lib 2>/dev/null \
+  || nix eval .#__std.direnv_lib # show the errors
+)"
+use std nix/cells //marlowe-playground/devshells:default
```
Comment (hrajchert) [inline .envrc:12]: What is this for?

File: cabal.project [hunk 2]
```diff
@@ -1,326 +1,46 @@
...
--- When updated to 3.12.0 or newer, remove the cabal-generator override in nix/pkgs/haskell/haskell.nix
-source-repository-package
-  type: git
-  location: https://github.com/input-output-hk/cardano-addresses
-  tag: 5094fb9d304ed69adedc99513634a00cbf850fca
-  --sha256: sha256-m8oidcV9LWU0S+NBw4wH1JdKZVict1EbKBZ6kfdEEf4=
-  subdir: command-line
-          core
-
-
```
Comment (hrajchert) [inline cabal.project:323]: 🙏🏻

File: nix/cells/marlowe-playground/operables.nix [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+      # FRONTEND_URL: The URL of the frontend server, for oauth redirection.
+
+      #################
+      # OPTIONAL VARS #
+      #################
+      # GITHUB_CLIENT_ID: The Client ID for the GitHub OAuth App
+      # GITHUB_CLIENT_SECRET: The Client secret for the GitHub OAuth App
+      # JWT_SIGNATURE: A secret signature for signing JWTs
+      # GITHUB_CALLBACK_PATH: A callback path for GitHub OAuth
+      # WEBGHC_URL: The URL of the WebGHC server.
```
Comment (hrajchert) [inline nix/cells/marlowe-playground/operables.nix:33]: How are these variables beign provided?

**Review by hrajchert — APPROVED** [review]
Code LGTM, local development still works, not sure how to test the deployment stuff, has this already deployed changes? or after merge?

File: .envrc (hunk 1 above)
Comment (jhbertra) [reply .envrc:12]: This is a file that `direnv` can use to automatically load the nix shell. It is for anyone who uses `direnv`

File: nix/cells/marlowe-playground/operables.nix (hunk 3 above)
Comment (jhbertra) [reply nix/cells/marlowe-playground/operables.nix:33]: The hosting infrastructure provides them. For example, we configure environment variables and secrets for them in KubeVela. A docker-compose manifest would specify them in the service's `env` config.

**Comment by jhbertra** [comment]
> Code LGTM, local development still works, not sure how to test the deployment stuff, has this already deployed changes? or after merge?

The app has already been deployed manually via the KubeVela UI. We are waiting on an integration from @input-output-hk/smartcontracts-developer-empowerment called `FluxCD` to enable automatic deployment from CI.

**Pull request merged**